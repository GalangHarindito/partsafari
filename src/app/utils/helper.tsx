import moment from "moment";

export const formatDate = (date: string) => {
    return moment(date).utcOffset(0).format('YYYY/MM/DD HH:mm')
}