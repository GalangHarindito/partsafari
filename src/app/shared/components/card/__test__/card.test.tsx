import { render, screen } from "@testing-library/react";
import Card from "../card";
import { formatDate } from "@/utils/helper";
import "@testing-library/jest-dom";

describe("Card Component", () => {
  const mockData = {
    id: 1,
    task: "Test Task",
    category: "Test Category",
    createdAt: "2023-10-01T12:00:00Z",
    completed: true,
  };

  test("Card layout unchanged", () => {
    const { container } = render(<Card data={mockData} />);
    expect(container).toMatchSnapshot();
  });

  test("render task and category", () => {
    render(<Card data={mockData} />);
    const taskElement = screen.getByText(mockData.task);
    const categoryElement = screen.getByText(mockData.category);
    expect(taskElement).toBeInTheDocument();
    expect(categoryElement).toBeInTheDocument();
  });

  test("renders checkbox with correct state", () => {
    render(<Card data={mockData} />);

    // Check if the checkbox is checked
    const checkbox = screen.getByRole("checkbox");
    expect(checkbox).toBeChecked();
  });

  test("render checkbox with false state", () => {
    const mockFalseCheck = { ...mockData, completed: false };
    render(<Card data={mockFalseCheck} />);

    const checkbox = screen.getByRole("checkbox");
    expect(checkbox).not.toBeChecked();
  });

  test("render date", () => {
    render(<Card data={mockData} />);

    const date = screen.getByText(formatDate(mockData.createdAt))
    expect(date).toBeInTheDocument();
  })
});
