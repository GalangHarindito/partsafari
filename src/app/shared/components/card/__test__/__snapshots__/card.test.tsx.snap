// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component Card layout unchanged 1`] = `
<div>
  <div
    class="bg-white grid grid-cols-[70px,1fr,2fr] p-2"
  >
    <div
      class="justify-self-center self-center"
    >
      <div
        class="flex items-center gap-2 cursor-pointer"
      >
        <label
          class="flex items-center justify-center w-4 h-4 relative"
        >
          <input
            checked=""
            class=" peer appearance-none shrink-0 w-4 h-4 border border-checkbox-gray-border rounded bg-white checked:bg-checkbox-blue "
            type="checkbox"
          />
          <svg
            class="absolute w-3.5 h-3.5 pointer-events-none hidden peer-checked:block stroke-white"
            fill="none"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2.3"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <polyline
              points="20 6 9 17 4 12"
            />
          </svg>
        </label>
      </div>
    </div>
    <div
      class="justify-self-start"
    >
      <h4
        class=" font-bold  xl:text-xl lg:text-lg text-base"
      >
        Test Task
      </h4>
      <p
        class="text-sm font-normal"
      >
        Test Category
      </p>
    </div>
    <div
      class="justify-self-end"
    >
      <p
        class="text-sm font-normal pt-1 pr-2"
      >
        2023/10/01 12:00
      </p>
    </div>
  </div>
</div>
`;
